driver-core/src/main/com/mongodb/connection/netty/NettyStreamFactoryFactory.java

```diff
@@ -99,6 +111,43 @@ public Builder eventLoopGroup(final EventLoopGroup eventLoopGroup) {
             return this;
         }
 
+        /**
+         * Sets a {@linkplain SslContextBuilder#forClient() client-side} {@link SslContext io.netty.handler.ssl.SslContext},
+         * which overrides the standard {@link SslSettings#getContext()}.
+         * By default it is {@code null} and {@link SslSettings#getContext()} is at play.
+         * <p>
+         * This option may be used as a convenient way to utilize
+         * <a href="https://www.openssl.org/">OpenSSL</a> as an alternative to the TLS/SSL protocol implementation in a JDK.
+         * To achieve this, specify {@link SslProvider#OPENSSL} TLS/SSL protocol provider via
+         * {@link SslContextBuilder#sslProvider(SslProvider)}. Note that doing so adds a runtime dependency on
+         * <a href="https://netty.io/wiki/forked-tomcat-native.html">netty-tcnative</a>, which you must satisfy.
+         * <p>
+         * Notes:
+         * <ul>
+         *    <li>Netty {@link SslContext} may not examine some
+         *    {@linkplain Security security}/{@linkplain System#getProperties() system} properties that are used to
+         *    <a href="https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#Customization">
+         *    customize JSSE</a>. Therefore, instead of using them you may have to apply the equivalent configuration programmatically,
+         *    if both the {@link SslContextBuilder} and the TLS/SSL protocol provider of choice support it.
+         *    </li>
+         *    <li>Only {@link SslProvider#JDK} and {@link SslProvider#OPENSSL} TLS/SSL protocol providers are supported.
+         *    </li>
+         * </ul>
+         *
+         * @param nettySslContext The Netty {@link SslContext}, which must be created via {@linkplain SslContextBuilder#forClient()}.
+         * @return {@code this}.
+         *
+         * @since 4.3
+         */
+        public Builder nettySslContext(final SslContext nettySslContext) {
```

Comment:
Naming convention question: Why `nettySslContext` and not just `SslContext`?

Reply:
I wanted to emphasize that Netty `SslContext` is not of type `javax.net.ssl.SSLContext`, however I was not sure whether it is a good idea to make names longer just for that purpose. I interpret your question as a sign that it was not a good idea.

Removed the `netty` prefix.

Done.